src/header/mod.rs

```diff
@@ -320,10 +320,11 @@ impl Headers {
     }
 
     /// Removes a header from the map, if one existed.
-    /// Returns true if a header has been removed.
-    pub fn remove<H: Header>(&mut self) -> bool {
+    /// Returns the header, if one has been removed.
```

Comment:
Hm, I don't love that the return value could be more complicated, but how about this crazy idea: `Option<Result<H, Raw>>`...

I'd personally just drop it, but is there a convincing case where the raw is still needed to justify complicating the return type?